hazelcast/src/main/java/com/hazelcast/map/impl/MapSizeEstimator.java

```diff
@@ -39,21 +44,34 @@ public void add(long size) {
 
     @Override
     public void reset() {
-        size = 0;
+        size = 0L;
     }
 
     @Override
-    public long getCost(T record) {
-        if (record == null) {
+    public long calculateSize(Object object) {
+        // currently we do not calculate heap cost for InMemoryFormat.OBJECT.
+        if (object == null || objectFormatUsed) {
             return 0L;
         }
-        final long cost = record.getCost();
-        if (cost == 0L) {
-            return cost;
+        final int refCost = 4;
```

Comment:
It is same with current calculation, calculations were designed to use 32-bit compressed oops. This may be subject of a different PR.